File: notifications/src/main/java/io/castled/notifications/trigger/BooleanConditionEvaluator.java
package io.castled.notifications.trigger;

import io.castled.notifications.exceptions.CastledRuntimeException;
import io.castled.notifications.trigger.models.PropertyOperation;
import io.castled.notifications.trigger.models.SingleValueOperation;

public class BooleanConditionEvaluator implements ParamsConditionEvaluator {

    @Override
    public boolean evaluateCondition(Object value, PropertyOperation propertyOperation) {
        Boolean aBool = (Boolean) value;
        Boolean conditionValue = Boolean.valueOf(((SingleValueOperation) propertyOperation).getValue());
        switch (propertyOperation.getOperationType()) {
            case EQ:
                return aBool != null && conditionValue.booleanValue() == aBool.booleanValue();
            default:
                throw new CastledRuntimeException(String.format("Operations type %s not supported for boolean operand", propertyOperation.getPropertyType()));
        }
    }
}

File: notifications/src/main/java/io/castled/notifications/trigger/BooleanConditionEvaluator.kt
package io.castled.notifications.trigger

import io.castled.notifications.trigger.enums.OperationType
import io.castled.notifications.trigger.models.PropertyOperation
import io.castled.notifications.trigger.models.SingleValueOperation

internal class BooleanConditionEvaluator : ParamsConditionEvaluator() {
    override fun evaluateCondition(value: Any?, propertyOperation: PropertyOperation): Boolean {
        val paramVal = value as Boolean?
        val conditionValue = ((propertyOperation as SingleValueOperation).value).toBoolean()

        return when (propertyOperation.type) {
            OperationType.EQ -> paramVal != null && paramVal == conditionValue
            else -> {
                logger.error("Operations type ${propertyOperation.propertyType} not supported for boolean operand")
                return false
            }
        }
    }
}